sdk/eventhubs/azure-messaging-eventhubs/src/private/eventhubs_utilities.hpp

```diff
@@ -277,7 +277,7 @@ namespace Azure { namespace Messaging { namespace EventHubs { namespace _detail
 #endif
       options.Properties.emplace(
           "user-agent",
-          Azure::Core::Http::_internal::UserAgentGenerator::GenerateUserAgent(
+          Azure::Core::Http::_internal::HttpShared::GenerateUserAgent(
```

Comment:
What's the thinking that led us to pick the `HttpShared` location for the home for this method?

Was it primarily because it already exists (so we don't have to come up with a new type/name)? That's a good reason.

Not pushing back on it, per-se, but trying to understand what we like about this spot.

Do we have any other existing spot, that doesn't have `Http` in the name? The reason being the expected usage is within other packages in the repo that explicitly are not using the HTTP protocol and policies (like EventHubs), so it's a bit odd to see code in those packages using a class called `HttpShared`.

Reply:
> Do we have any other existing spot, that doesn't have `Http` in the name? The reason being the expected usage is within other packages in the repo that explicitly are not using the HTTP protocol and policies (like EventHubs), so it's a bit odd to see code in those packages using a class called `HttpShared`.

Nevermind, I just searched and saw a bunch of hits for `Azure::Core::Http` within EventHubs, including RetryOptions.

Reply:
Yes, it is the fact that it already exists. It may not the best, but it is there already.